cpp/src/arrow/filesystem/gcsfs_internal.h

```diff
@@ -30,6 +32,15 @@
 
 namespace arrow {
 namespace fs {
+struct GcsCredentialsHolder {
```

Comment:
```suggestion
struct ARROW_EXPORT GcsCredentialsHolder {
```

Reply:
I am not sure we want to do this.  That symbol was not exported before, and was not part of the public API:

https://github.com/apache/arrow/blob/cd6b2b5dca468cee6704ade4ce539c0a09874607/cpp/src/arrow/filesystem/gcsfs.cc#L38-L43

arguably it should live in `arrow::fs::internal`.  Thoughts?

Reply:
> it should live in `arrow::fs::internal`. 

Ah, yes. I agree with it.

Reply:
Done

Reply:
Thanks!